rust/src/lib.rs

```diff
@@ -716,8 +717,56 @@ pub unsafe extern "C" fn zcashlc_get_current_address(
     unwrap_exc_or_null(res)
 }
 
+bitflags! {
+    /// A set of bitflags used to specify the types of receivers a unified address can contain. The
+    /// flag bits chosen here for each receiver type are incidentally the same as those used for
+    /// serialization in `zcash_client_sqlite`; consistency here isn't really meaningful but is
+    /// less confusing than letting them diverge.
+    #[derive(Clone, Copy, Debug, PartialEq, Eq)]
+    struct ReceiverFlags: u32 {
+        /// The requested address can receive transparent p2pkh outputs.
+        const P2PKH = 0b00000001;
+        /// The requested address can receive Sapling outputs.
+        const SAPLING = 0b00000100;
+        /// The requested address can receive Orchard outputs.
+        const ORCHARD = 0b00001000;
+    }
+}
+
+impl ReceiverFlags {
+    fn to_address_request(&self) -> Result<UnifiedAddressRequest, ()> {
+        UnifiedAddressRequest::custom(
+            if self.contains(ReceiverFlags::ORCHARD) {
+                ReceiverRequirement::Require
+            } else {
+                ReceiverRequirement::Omit
+            },
+            if self.contains(ReceiverFlags::SAPLING) {
+                ReceiverRequirement::Require
+            } else {
+                ReceiverRequirement::Omit
+            },
+            if self.contains(ReceiverFlags::P2PKH) {
+                ReceiverRequirement::Require
+            } else {
+                ReceiverRequirement::Omit
+            },
+        )
+    }
+}
+
 /// Returns a newly-generated unified payment address for the specified account, with the next
-/// available diversifier.
+/// available diversifier and the specified set of receivers.
+///
+/// The set of receivers to include in the generated address is specified by a byte which may have
+/// any of the following bits set:
+/// * P2PKH = 0b00000001
+/// * SAPLING = 0b00000100
+/// * ORCHARD = 0b00001000
```

Comment:
Nit: now that the bit length is not `u8`, these might be clearer either with the full set of prefixing zeroes (so there is no possible ambiguity in where the extra zero bits go), or as `(1 << 0)` etc. so we're just stating the relevant bit position.